app/components/markdown.tsx

```diff
@@ -100,20 +100,20 @@ export function PreCode(props: { children: any }) {
 }
 
 function escapeDollarNumber(text: string): string {
-  let isInMultilineCode = false;
+  let isInBlockCode = false;
   return text.split('\n').map(line => {
     if (line.trim() === '```' || line.trim() === '`') {
-      isInMultilineCode = !isInMultilineCode;
+      isInBlockCode = !isInBlockCode;
       return line;
     }
-    if (!isInMultilineCode) {
-      return line.replace(/(`[^`]*`)|(\$[0-9]+(?:[,.][0-9]+)*)/g, (match, inlineCode, number) => {
-        if (inlineCode) {
-          return inlineCode;
+    if (!isInBlockCode) {
+      return line.split(/(`.*?`)/g).map((segment, index) => {
+        if (index % 2 === 0) {
```

Comment:
now it only handle correctly for left but not the right one

![image](https://github.com/ChatGPTNextWeb/ChatGPT-Next-Web/assets/17626300/77ed40be-71ac-4aa7-9edd-0981e31cc5e6)